### PR TITLE
[Improvement] Fix the git url command injection in pytorch task(#15873)

### LINKED ISSUE #15873: [Bug] [Pytorch] There is no security check for GitProjectManager.getGitUrl, which could cause command injection
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

https://github.com/apache/dolphinscheduler/blob/8fc204940f7be5cf356f63dd223bc599f857fe69/dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/main/java/org/apache/dolphinscheduler/plugin/task/pytorch/GitProjectManager.java#L47-L50
There is no input validation for `getGitUrl` and users could put shell commands here.

### What you expected to happen

ensure the  `getGitUrl` is a valid url

### How to reproduce

image
image

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by cntigers**
## Purpose of the pull request

fix: [15873](https://github.com/apache/dolphinscheduler/issues/15873)

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java [hunk 1]
```diff
@@ -72,6 +72,12 @@ public void testPythonEnvManager() {
 
     }
 
+    @Test
+    public void testGitProjectUrlInjection(){
+        Assertions.assertFalse(GitProjectManager.isGitPath("git@& cat /etc/passwd >/poc.txt #"));
+        Assertions.assertFalse(GitProjectManager.isGitPath("git@| cat /etc/passwd >/poc.txt #"));
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java:78]: Please add a positive case which will return true.

Comment (cntigers) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java:78]: > Please add a positive case which will return true.

Thank you for your reminder. I found that there are already positive test cases during testing @ruanwenjun 

 [exists test cases](https://github.com/apache/dolphinscheduler/blob/d218b021e7e045e7982a3c91ea7b1b1491d21d38/dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java#L76)

Comment (ruanwenjun) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java:78]: Great, please use `mvn spotless:apply` to format the code style.

Comment (cntigers) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-pytorch/src/test/java/org/apache/dolphinscheduler/plugin/task/pytorch/PytorchTaskTest.java:78]: > Great, please use `mvn spotless:apply` to format the code style.

Thank you for your advice,I have already resolved it.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by wangxj3 — APPROVED** [review]
LGTM

**Review by rickchengx — APPROVED** [review]
LGTM

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**